storage/objects/main.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Sample objects creates, list, deletes objects and runs
+// other similar operations on them by using the Google Storage API.
+// More documentation is available at
+// https://cloud.google.com/storage/docs/json_api/v1/.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"cloud.google.com/go/storage"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	if projectID == "" {
+		fmt.Fprintf(os.Stderr, "GOOGLE_CLOUD_PROJECT environment variable must be set.\n")
+		os.Exit(1)
+	}
+	var o string
+	flag.StringVar(&o, "o", "", "source object; in the format of <bucket:object>")
+	flag.Parse()
+
+	names := strings.Split(o, ":")
+	if len(names) < 2 {
+		usage("missing -o flag")
+	}
+	bucket, object := names[0], names[1]
+
+	if len(os.Args) < 2 {
+		usage("missing subcommand")
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
```

Comment:
The tutorials prepared for these snippets are very independent. Without giving enough context about how BucketHandle and ObjectHandle are constructed, it is very hard for the developers to follow. That's why every snippet is beginning at the client level.